File: apps/web/types/language.ts
export interface Language {
  id: string;
  user_id: string;
  name: string;
  level: string | null;
  created_at: string;
  updated_at: string;
}

export interface LanguageCreate {
  name: string;
  level?: string | null;
}

export type LanguageUpdate = Partial<LanguageCreate>;

File: apps/api/app/schemas/language.py
"""Language schemas"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class LanguageBase(BaseModel):
    name: str = Field(..., max_length=100)
    level: Optional[str] = Field(None, max_length=100)


class LanguageCreate(LanguageBase):
    pass


class LanguageUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=100)
    level: Optional[str] = Field(None, max_length=100)


class LanguageResponse(LanguageBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

